sycl/doc/extensions/proposed/sycl_ext_oneapi_improved_host_task.asciidoc

```diff
@@ -0,0 +1,318 @@
+more properties can be provided to the enqueuing of the host task command via an
+instance of `property_list`.
+
+[NOTE]
+====
+Normally a `property_list` parameter would be added with a default argument,
+however as `host_task` did not originally take a `property_list` parameter doing
+this would cause an ambiguity.
+====
+
+
+=== Retrieving or adding events in a host task
+
+New member functions are added to the `interop_handle` class for retrieving
+dependent native events when the `property::host_task::manual_interop_sync`
+property is used when submitting the host task.
+
+Additionally new member functions are added to the `interop_handle` class for
+adding native events created within the host task. These member functions add
+native events as dependencies to the SYCL `event` returned from the submission
+of the host task as dependent events.
+
+[source,c++]
+----
+template <backend Backend>
+std::vector<backend_return_t<Backend, event>> ext_oneapi_get_native_events();
+----
+
+_Effects_: Returns a `std::vector` of the native events for the backend
+`Backend` for the dependencies for the dependencies to the host task command if
+the host task command was enqueued with the
+`property::host_task::manual_interop_sync` property, otherwise returns an
+empty `std::vector`.
+
+[source,c++]
+----
+template <backend Backend>
+void ext_oneapi_add_native_events(
+  backend_return_t<Backend, event> hostTaskEvent);
+
+template <backend Backend>
+void ext_oneapi_add_native_events(
+  const std::vector<backend_return_t<Backend, event>> &hostTaskEvents)
+----
+
+_Effects_: Adds the native event(s) `hostTaskEvent`/`hostTaskEvents` as an 
+additional dependency to the host task command completion, that will be waited
+on after waiting on the invocation of the host task function any time where the
+host task function would be waited on including `event::wait`, `queue:wait` and
+`queue::wait_and_throw`. If the function is called multiple times all native
+events provided will be waited on collectively.
+
+[NOTE]
+====
+The `std::vector` returned may also be empty if there are no dependencies for
+which there is a native event for the backend.
+====
+
+
+== Example
+
+Below is an example of using the new interfaces using the OpenCL backend.
+
+[source,c++]
+----
+int pattern = 42;
+
+auto e1 = queue.submit([=](sycl::handler &cgh) {
+  accessor acc{bufA, cgh};
+ 
+  cgh.parallel_for<kernelA>([=](sycl::id<1> idx) {
+    acc[0] = 2;
+  });
+});
+
+auto e2 = queue.submit([&](sycl::handler &cgh) {
+  accessor acc{bufB, cgh};
+
+  // creates a dependency on the previous kernel execution
+  cgh.depends_on(e1);
+
+  auto manualInteropSync =
+    ext::oneapi::experimental::property::host_task::manual_interop_sync;
+
+  cgh.host_task([&](sycl::interop_handle &ih, {ext::oneapi::experimental}) {
+    // Dependent events are returned to be synchronized with.
+    auto nativeEvents = ih.get_native_events<backend::opencl>();
```

Comment:
where do nativeEvents come from, from line 178 `ih.ext_oneapi_add_native_events<backend::opencl>(ne2);` of the last host_task?

If the answer is yes, how about `host_task1` --> `sycl kernel` --> `host_task2`,  actually, the `host_task2` does not depend on the `host_task1`, but depends on `sycl kernel`, so, we don't need the waiting events in line 270, the dependency is handled in line 256.

Even for the case `host_task1` --> `host_task2`, since the native event in `host_task1` is wrapped as sycl::event, and so for `host_task2`, line 256 is enough, we don't need the waiting events at line 270.

Looks that get_native_events is not necessary.